feast_trino/trino.py

```diff
@@ -65,12 +65,15 @@ def on_demand_feature_views(self) -> Optional[List[OnDemandFeatureView]]:
     def _to_df_internal(self) -> pd.DataFrame:
         """Return dataset as Pandas DataFrame synchronously including on demand transforms"""
         results = self._client.execute_query(query_text=self._query)
-        return pd.DataFrame(data=results.data, columns=results.columns_names)
+        df = pd.DataFrame(data=results.data, columns=results.columns_names)
+        for col_name, col_type in results.schema.items():
+            if col_type.startswith("timestamp"):
+                df[col_name] = pd.to_datetime(df[col_name])
+        return df
 
     def _to_arrow_internal(self) -> pyarrow.Table:
         """Return payrrow dataset as synchronously including on demand transforms"""
-        results = self._client.execute_query(query_text=self._query)
-        return pyarrow.table(data=results.data, columns=results.columns_names)
+        return pyarrow.Table.from_pandas(self._to_df_internal())
```

Comment:
This is going to be tested multiple times once https://github.com/Shopify/feast-trino/pull/22 is merged (and so we will run the universal tests coming from Feast)